### DISCUSSION TITLE
Is there a way to make `ruff format` report examples with invalid syntax in docstrings?

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#13622] [Q&A] [answered]
Is there a way to make `ruff format` detect examples with invalid syntax in docstrings?

https://docs.astral.sh/ruff/formatter/#docstring-formatting

> If a code example is recognized and treated as Python, the Ruff formatter will automatically skip it if the code does not parse as valid Python or if the reformatted code would produce an invalid Python program.

**Comment by MichaReiser** [comment, marked as answer]
No, this is currently not implemented, mainly because the formatter does not support emitting any form of diagnostics. CC: @BurntSushi for another diagnostics use case in the formatter other than diff diagnostics.

**Reply by harupy** [reply]
@MichaReiser Thanks for the clarification!
